auditbeat/module/auditd/config_test.go

```diff
@@ -265,7 +265,7 @@ func parseConfig(t testing.TB, yaml string) (Config, error) {
 }
 
 func commands(rules []auditRule) []string {
-	var cmds []string
+	var cmds []string //nolint:prealloc // github.com/elastic/beats/pull/37063#discussion_r1385580086
```

Comment:
Here prealloc is fine. The comment you have linked to; there Dan is right. But this is different; here you know how much to preallocate. You can see this [answer](https://stackoverflow.com/a/25543590/5821408) on SO to know why `var` is better than initialize with length 0.

`cmd := make([]string, 0, len(rules)`

Reply:
Don't we also know the length in the other two instances? The loop can either go to `len(elements)` or the function will return an error in both cases so in theory it should be ok to preallocate

Reply:
I have commented on one of the two cases. As for the other case, which is the 'hashes' case, I suggest preallocating with `len(hasher.config.HashTypes)`, though to be honest, there hardly would be any perf gain. `hasher.config.HashTypes` is configured from the config, and ideally, it should only go up to 16 (see: [validHashes](https://github.com/elastic/beats/blob/df0d35c2567a49f1f4eb898a17d0deed45d93fc5/auditbeat/helper/hasher/hasher.go#L58)). However, if the user provided duplicates in the config, then it will add up, but this is very unlikely. Therefore, given that it can go up to just 16, preallocating with `len(hasher.config.HashTypes)` or leaving it as it is will suffice.

Reply:
> As for the other case, which is the 'hashes' case

I am confused about what has been said in the comments :D, I'll leave the current code and comment it

Reply:
I was talking about this one: https://github.com/elastic/beats/blob/52e9f61dc4d4b62e2fefdc43aa9415e22d8f3817/auditbeat/helper/hasher/hasher.go#L202